android/java/org/chromium/chrome/browser/rewards/onboarding/RewardsOnboarding.java

```diff
@@ -0,0 +1,303 @@
+        responseActionButton.setOnClickListener(v -> {
+            mBraveRewardsNativeWorker.getAvailableCountries();
+            mLocationChooseLayout.setVisibility(View.VISIBLE);
+            mErrorLayout.setVisibility(View.GONE);
+        });
+        View responseCloseButton = mErrorLayout.findViewById(R.id.response_modal_close);
+        responseCloseButton.setOnClickListener(v -> { mPopupWindow.dismiss(); });
+
+        mPopupWindow.setOnDismissListener(new PopupWindow.OnDismissListener() {
+            @Override
+            public void onDismiss() {
+                if (mBraveRewardsNativeWorker != null) {
+                    mBraveRewardsNativeWorker.RemoveObserver(RewardsOnboarding.this);
+                }
+            }
+        });
+
+        mPopupWindow.setWidth(deviceWidth);
+        mPopupWindow.setContentView(mPopupView);
+
+        mBraveRewardsNativeWorker = BraveRewardsNativeWorker.getInstance();
+        mBraveRewardsNativeWorker.AddObserver(this);
+    }
+
+    public void showLikePopDownMenu() {
+        mPopupWindow.setTouchable(true);
+        mPopupWindow.setFocusable(true);
+        mPopupWindow.setOutsideTouchable(true);
+        mPopupWindow.showAsDropDown(mAnchorView, 0, 0);
+    }
+
+    private void showRewardsTour() {
+        mPopupWindow.dismiss();
+        TabUtils.openUrlInNewTab(false, BraveRewardsPanel.REWARDS_TOUR_URL);
+    }
+
+    @Override
+    public void onGetAvailableCountries(String[] countries) {
+        View progressBar = mPopupView.findViewById(R.id.progressBar);
+        progressBar.setVisibility(View.GONE);
+        updateCountryList(countries);
+    }
+
+    private void updateCountryList(String[] countries) {
+        shouldShowContinueProgress(false);
+        mContinueButton.setText(mActivity.getResources().getString(R.string.continue_text));
+        TreeMap<String, String> sortedCountryMap = new TreeMap<String, String>();
+        for (String countryCode : countries) {
+            sortedCountryMap.put(new Locale("", countryCode).getDisplayCountry(), countryCode);
+        }
+
+        ArrayList<String> countryList = new ArrayList<String>();
+        countryList.add(mActivity.getResources().getString(R.string.select_your_country_title));
+        countryList.addAll(sortedCountryMap.keySet());
+        String defaultCountry = mBraveRewardsNativeWorker.getCountryCode() != null
+                ? new Locale("", mBraveRewardsNativeWorker.getCountryCode()).getDisplayCountry()
+                : null;
+        if (defaultCountry != null && countryList.contains(defaultCountry)) {
+            countryList.remove(defaultCountry);
+            countryList.add(1, defaultCountry);
+        }
+
+        String[] countryArray = countryList.toArray(new String[countryList.size()]);
+        Spinner countrySpinner;
+        countrySpinner = mPopupView.findViewById(R.id.country_spinner);
+
+        // if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
we should remove commented code here if we are not planning to use it.

Reply:
updated removed commented code